src/dsp_tools/commands/xmlupload/models/ontology_diagnose_models.py

```diff
@@ -81,29 +90,42 @@ def _get_problems_as_df(self) -> pd.DataFrame:
         unpacked: list[dict[str, str]] = list(itertools.chain(*problems))
         return pd.DataFrame.from_records(unpacked)
 
+    def _calculate_num_resources(self, to_count: list[tuple[str, list[str], str]]) -> int:
+        return sum((len(x[1]) for x in to_count))
+
     def _compose_problem_string_cls(self) -> str | None:
```

Comment:
My first thought was that this name means "compose a problem string for this class". It took me really long to understand that there is a corresponding method `_compose_problem_string_props`, and that one method does it for classes, the other for props. 

Now I see that in my mind, I have to add a "for" or a "of" before "cls".

I would rename this function to `_compose_problem_string_from_classes` or `_compose_problem_string_of_cls`. 

And then the same for `_compose_problem_string_props`